### PR TITLE
Use kubeadm alpha certs generate-csr in demo

### PR DESCRIPTION
**Pull request by wallrj**
Use the new `kubeadm alpha certs generate-csr` command from https://github.com/kubernetes/kubernetes/pull/92183 to simplify the creation of CSR files and to also create CSR and sign certificates for kube config files.

Signed-off-by: Richard Wall <richard.wall@jetstack.io>

### CODE REVIEW

File: docs/demos/kubeadm-bootstrap/README.md [hunk 1]
```diff
@@ -308,41 +336,12 @@ This is because we are using the same CA for signing the `Etcd` and [Aggregated
...
-kubeadm init phase kubeconfig all \
-           --cert-dir "${PWD}/pki.self-signed" \
-           --kubeconfig-dir "kubernetes/" \
-           --control-plane-endpoint kind-control-plane \
-           --node-name kind-control-plane
-```
-
-Notice that we set the API server hostname and the kubelet node name to `kind-control-plane` to match the name used by [Kind].
-
+These `kubeconfig` files will have been generated by the `kubeadm alpha certs generate-csr` above, but you now need to update those files with the
```
Comment (hzhou97) [inline docs/demos/kubeadm-bootstrap/README.md]: with the ?

File: docs/demos/kubeadm-bootstrap/kubeadm-bootstrap.sh [hunk 2]
```diff
@@ -24,29 +25,20 @@ function log() {
...
-        policy="$(echo "${nickname}" | grep -o '\(client\|server\|peer\)$')"
+        policy="$(echo "${nickname}" | grep -o '\(client\|server\|peer\)$' || true)"
         # Special case for etcd-server which needs both server and client usage
         # See https://clusterise.com/articles/kbp-2-certificates/ and
         # https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
         if [[ "${nickname}" == "etcd-server" ]]; then
             policy=peer
         fi
-        log "Enrolling CSR ${csr} with nickname ${nickname} to ${cert}"
+        if echo "${nickname}" | grep '\.conf$' > /dev/null; then
```
Comment (hzhou97) [inline docs/demos/kubeadm-bootstrap/kubeadm-bootstrap.sh:41]: use `if [[ "${nickname}" == *\.conf ]]`?
Not too sure about bash syntax actually, have a look at https://www.diskinternals.com/linux-reader/bash-string-ends-with/

File: docs/demos/kubeadm-bootstrap/README.md [hunk 3]
```diff
@@ -68,57 +77,74 @@ and save it as `ca.venafi.crt`.
...
 
 ## Generating Certificate Signing Requests
 
-[Kubeadm] allows you to perform individual phases of its Kubernetes control-plane creation sequence.
-Use the sub-command `kubeadm init --help` to see an overview of these phases and
-use `kubeadm init phase --help` to see how to run a single phase.
-
-Run sub-steps of the `certs` phase to generate certificate signing requests for each of the Kubernetes control-plane components,
-which we will later submit to TPP using the `vcert` command.
+Run `kubeadm alpha certs generate-csr`to generate private keys and CSR files for all the certificates required by the Kubernetes control-plane.
```
Comment (hzhou97) [inline docs/demos/kubeadm-bootstrap/README.md]: space missing before "to generate"

File: docs/demos/kubeadm-bootstrap/README.md [hunk 4]
```diff
@@ -188,34 +214,36 @@ Here is a snippet of bash which can be used to sign all the certificates:
...
-        policy="$(echo "${nickname}" | grep -o '\(client\|server\|peer\)$')"
+        policy="$(echo "${nickname}" | grep -o '\(client\|server\|peer\)$' || true)"
         # Special case for etcd-server which needs both server and client usage
         # See https://clusterise.com/articles/kbp-2-certificates/ and
         # https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates
         if [[ "${nickname}" == "etcd-server" ]]; then
             policy=peer
         fi
-        log "Enrolling CSR ${csr} with nickname ${nickname} to ${cert}"
+        if echo "${nickname}" | grep '\.conf$' > /dev/null; then
```
Comment (hzhou97) [inline docs/demos/kubeadm-bootstrap/README.md:224]: If you change the script, do it here as well.

**Comment by wallrj** [comment]
Thanks for the review @hzhou97 I didn't see  your comments until just now. 
I fixed a couple of typos but left the bash syntax (for now) for fear of breaking things after I'd just recorded a new screencast.
I'll try and improve the scripts in later PRs

**Pull request merged**